### PR TITLE
amazon: allow NACL to be removed by its id

### PR DESCRIPTION
**Pull request by ghost**
##### SUMMARY
**ec2_vpc_nacl_facts** can list NACLs without *Name* tag, but **ec2_vpc_nacl** cannot delete them. Fixing this.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
cloud/amazon/ec2_vpc_nacl

##### ANSIBLE VERSION
```
ansible 2.4.0 (nacl 3778f56cf3) last updated 2017/06/20 21:16:19 (GMT +200)
  config file = None
  configured module search path = [u'/home/xxx/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/xxx/work/ansible/lib/ansible
  executable location = bin/ansible
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]
```


##### ADDITIONAL INFORMATION
Allows to delete NACLs with missing Name tag
```
- name: "Delete nacl by it's id"                                                                                                                                                                                                                                                
  ec2_vpc_nacl:                                                                                                                                                                                                                                                                 
    nacl_id: acl-33b4ee5b                                                                                                                                                                                                                                                       
    state: absent 
```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py [hunk 1]
```diff
@@ -552,6 +571,12 @@ def main():
         "present": setup_network_acl,
         "absent": remove_network_acl
     }
+    if state == 'present' and not module.params.get('vpc_id'):
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py]: Use required_if in AnsibleModule for this:

```
required_if=[['state', 'present', ['vpc_id']]]
```

File: lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py [hunk 2]
```diff
@@ -552,6 +571,12 @@ def main():
...
+        module.fail_json(msg="Require vpc_id when state='present'.")
+
+    if not module.params.get('name') and not module.params.get('nacl_id'):
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py]: Use `required_one_of` for this:

```
required_one_of=[['name', 'nacl_id']]
```

File: lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py [hunk 3]
```diff
@@ -527,8 +545,9 @@ def subnets_to_associate(nacl, client, module):
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
-        vpc_id=dict(required=True),
-        name=dict(required=True),
+        vpc_id=dict(required=False),
+        name=dict(required=False),
+        nacl_id=dict(required=False),
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py]: `required=False` is the default, so these can just be empty `dict()` parameters

File: lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py [hunk 4]
```diff
@@ -118,6 +126,11 @@
     vpc_id: vpc-12345678
     name: prod-dmz-nacl
     state: absent
+
+- name: "Delete nacl by it's id"
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py]: The apostrophe in `it's` is incorrect here.

File: lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py [hunk 5]
```diff
@@ -29,11 +29,19 @@
   name:
     description:
       - Tagged name identifying a network ACL.
-    required: true
+      - One of the name or nacl_id is required.
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py]: Probably should be 'One and only one of name and nacl_id is required'

Comment (willthames) [reply lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py]: Also, use name quoting - in this case, as they're option names, it should be `I(name) and I(nacl_id)`

**Review by willthames — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@willthames thanks for the quick review. Fixes pushed.

**Review by willthames — APPROVED** [review]

**Comment by willthames** [comment]
Merged, thanks

**Pull request merged**